data_diff/dbt.py

```diff
@@ -230,14 +232,24 @@ def _initialize_api() -> Optional[DatafoldAPI]:
 
     api_key = os.environ.get("DATAFOLD_API_KEY")
     if not api_key:
-        rich.print("[red]API key not found, add it as an environment variable called DATAFOLD_API_KEY.")
-        yes_or_no = Confirm.ask("Would you like to generate a new API key?")
-        if yes_or_no:
-            webbrowser.open(f"{datafold_host}/login?next={datafold_host}/users/me")
-            rich.print('After generating, please, perform in the terminal "export DATAFOLD_API_KEY=<key>"')
-            return None
-        else:
-            raise ValueError("Cannot initialize API because the API key is not provided")
+        rich.print("[red]API key not found. Trying to get from the system keyring service")
+        api_key = keyring.get_password("data-diff", "DATAFOLD_API_KEY")
+        if not api_key:
+            rich.print("[red]API key not found, add it as an environment variable called DATAFOLD_API_KEY.")
+
+            yes_or_no = Confirm.ask("Would you like to generate a new API key?")
+            if yes_or_no:
+                webbrowser.open(f"{datafold_host}/login?next={datafold_host}/users/me")
+                rich.print('After generating, please, perform in the terminal "export DATAFOLD_API_KEY=<key>"')
+                return None
+            else:
+                raise ValueError("Cannot initialize API because the API key is not provided")
+
+    rich.print("Trying to save or update the API key to the system keyring service")
```

Comment:
If we use the word "trying" we should have a success message

Otherwise we could make this message more like "Saving the API key to the system keyring service"

Reply:
I have just followed the logic that if all is fine we do not need "success" here but if you think it is worth to change it, no problem, i will update.